google/cloud/bigtable/instance_admin.cc

```diff
@@ -33,209 +25,82 @@ namespace google {
 namespace cloud {
 namespace bigtable {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
 static_assert(std::is_copy_assignable<bigtable::InstanceAdmin>::value,
               "bigtable::InstanceAdmin must be CopyAssignable");
 
-using ClientUtils = bigtable::internal::UnaryClientUtils<InstanceAdminClient>;
-using ::google::cloud::Idempotency;
-
 StatusOr<InstanceList> InstanceAdmin::ListInstances() {
-  grpc::Status status;
+  google::cloud::internal::OptionsSpan span(policies_);
   InstanceList result;
-  // Copy the policies in effect for the operation.
-  auto rpc_policy = clone_rpc_retry_policy();
-  auto backoff_policy = clone_rpc_backoff_policy();
 
   // Build the RPC request, try to minimize copying.
-  std::unordered_set<std::string> unique_failed_locations;
-  std::string page_token;
-  do {
-    btadmin::ListInstancesRequest request;
-    request.set_page_token(std::move(page_token));
-    request.set_parent(project_name());
-
-    auto response = ClientUtils::MakeCall(
-        *(client_), *rpc_policy, *backoff_policy,
-        MetadataUpdatePolicy(project_name(), MetadataParamTypes::PARENT),
-        &InstanceAdminClient::ListInstances, request,
-        "InstanceAdmin::ListInstances", status, Idempotency::kIdempotent);
-    if (!status.ok()) {
-      break;
-    }
-
-    auto& instances = *response.mutable_instances();
-    std::move(instances.begin(), instances.end(),
-              std::back_inserter(result.instances));
-    auto& failed_locations = *response.mutable_failed_locations();
-    std::move(
-        failed_locations.begin(), failed_locations.end(),
-        std::inserter(unique_failed_locations, unique_failed_locations.end()));
-
-    page_token = std::move(*response.mutable_next_page_token());
-  } while (!page_token.empty());
-
-  if (!status.ok()) {
-    return MakeStatusFromRpcError(status);
-  }
-
-  std::move(unique_failed_locations.begin(), unique_failed_locations.end(),
+  btadmin::ListInstancesRequest request;
+  request.set_parent(project_name());
+  auto sor = connection_->ListInstances(request);
```

Comment:
The previous code iterates over the results, possibly making multiple calls. I do not think `Connection` does that?

Reply:
(responding only to this comment) `ListInstances` is no longer a paginated api. See: https://github.com/googleapis/googleapis/blob/853e8a29cecd0816c8b003b299032c3a5168ef8b/google/bigtable/admin/v2/bigtable_instance_admin.proto#L347-L348

This is also true of `ListClusters`: https://github.com/googleapis/googleapis/blob/853e8a29cecd0816c8b003b299032c3a5168ef8b/google/bigtable/admin/v2/bigtable_instance_admin.proto#L436-L437